_posts/2017-08-16-json-server.md

```diff
@@ -0,0 +1,655 @@
+---
+layout: post
+title: JSON Server
+excerpt: Accelerate your prototyping process by mocking your APIs using JSON Server
+authors:
+    - kelfarsaoui
+permalink: /en/json-server/
+categories:
+    - JSON
+    - API
+    - NodeJS
+tags:
+    - JSON
+    - API
+    - NodeJS
+cover: /assets/2017-08-16-json-server/cover.jpg
+---
+
+Hello, Today I’m going to talk about [json-server](https://github.com/typicode/json-server), what is it? why use it? and especially how to use it?
+
+`json-server` is a cool `npm` module, that provides an [Express](https://github.com/expressjs/express) server that serves a JSON API.
+
+## Why use it?
+
+Let's say you are starting to work on your awesome (Javascript, PHP, IOS, Android, … whatever) app, and you want to consume remote data of some, not finished yet, API. The first step you take consists in setting up some mocks, either hard coded in some constant (please, don't do this), or using a static JSON file, which will be a pain in the neck. Why? You're a good developer who likes to do the right thing. You want to test your app against data updates. You want to perform HTTP requests (like `GET`, `POST` …etc.), and you want to persist your updates. Unfortunately, you can't do all of this using a static file, you need to look for a way to make it dynamic. So, unless your pal who's developing the API has finished their work, you're gonna need some serious help.
```

Comment:
Let's say you are starting to work on your awesome (Javascript, PHP, IOS, Android, … whatever) app, and you want to consume remote data of some API which is not finished yet. The first step you take consists in setting up some mocks, either hard coded in some constant (please, don't do this), or using a static JSON file, which will be a pain in the neck. Why? You are a good developer who likes to do the right thing : you want to test your app against data updates, you want to perform HTTP requests (like `GET`, `POST` …etc.), and you want to persist your updates. Unfortunately, you can't do all of this by using a static file, you need to look for a way to make it dynamic. So, unless your pal who's developing the API has finished his work, you're gonna need some serious help.